### PR TITLE
Fix milestones order issue

### LINKED ISSUE #8593: [BUG] Milestones date sorting
### Description

Dates are sorted incorrectly on the Milestones. 
image

Milestones should be sorted from oldest to newest (or whatever it is) but it is sorted by month. If I have milestones in December they will appear as the most recent, instead of the most recent year being the most recently like it probably should.

### Screenshots

image

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Giuliano1993**
## Fixes 

Closes #8593

## Changes proposed

- In the public profile, ordering by date for milestones was missing; I added it to the mongo call to fix the issue

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

Screenshot 2023-09-02 201823
Screenshot 2023-09-02 201758

## Note to reviewers
NB: Even using the same dates from the risen issue, I could reproduce the issue only on the public profile, while managing your milestone didn't present any problem, the order was correct.

### CODE REVIEW

File: pages/api/profiles/[username]/index.js [hunk 1]
```diff
@@ -51,6 +51,16 @@ export async function getUserApi(req, res, username, options = {}) {
     {
       $match: { username },
     },
+    {
+      $set:{
+        "milestones":{
+          $sortArray:{
+            input:"$milestones",
+            sortBy: {date:1}
```
Comment (sital002) [inline pages/api/profiles/[username]/index.js:59]:
```suggestion
            sortBy: {date:-1}
```
As discussed on the issue it should show the most recent ones.
image

**Comment by Giuliano1993** [comment]
Yes @sital002 sorry, I missed that point at first. Shall I apply the same changes to the account section too then?

File: pages/api/profiles/[username]/index.js [hunk 2]
```diff
@@ -51,6 +51,16 @@ export async function getUserApi(req, res, username, options = {}) {
...
+    {
+      $set:{
+        "milestones":{
+          $sortArray:{
+            input:"$milestones",
+            sortBy: {date:-1}
+          }
+        }
+      }
+    },
```
Comment (eddiejaoude) [inline pages/api/profiles/[username]/index.js:63]: Formatting ...

```suggestion
    {
      $set: {
        milestones: {
          $sortArray: {
            input: "$milestones",
            sortBy: { date: -1 },
          },
        },
      },
    },
```

**Review by eddiejaoude — APPROVED** [review]
Looks great 👍

**Pull request merged**